include/hlslib/xilinx/DataPack.h

```diff
@@ -16,6 +17,96 @@ class DataPackProxy; // Forward declaration
 
 } // End anonymous namespace
 
+/// Helper class to allow more efficient packing on the FPGA, where memory
+/// access is not restricted to byte-sized chunks.
+/// Specializations of this class for
+template <typename T>
+class TypeHandler {
```

Comment:
I would make this a `struct` (then stuff is public by default), and maybe stick it in a `namespace detail` (this seems to be a typical convention for wrapping helper types that are not supposed to be used in any other context).

Reply:
Good call, will do!